# The California Wildfire Damage Prediction

# Introduction

Wildfires in California have become more frequent and severe, causing significant environmental and economic damage. This project aims to predict wildfire damage using machine learning by analyzing historical wildfire data, weather conditions, and geographical features.

# Description:
The dataset reflects the damage sustained by structures across various fire incidents, categorized by damage percentage—ranging from minor damage (1-10%) to complete destruction (50-100%) and collected by field inspectors who evaluate structures impacted by wildland fires.

# Objective:
The goal is to develop a Reggression classifier or suitable ML model  to predict wildfire damage in California using historical data, weather conditions, and geographical factors. By analyzing fire patterns, cleaning data, and building predictive models, this project aims to enhance early warning systems, improve resource allocation, and support risk assessment for wildfire managemen

# Loading dataset

In [15]:
# import packages and dataset loading
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

df = pd.read_csv('/content/The California Wildfire Data.csv', encoding='ISO-8859-1')
df.head()

<ipython-input-15-c5d7ff2ec42a>:6: DtypeWarning: Columns (13,37,38) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('/content/The California Wildfire Data.csv', encoding='ISO-8859-1')


_id  OBJECTID         * Damage  * Street Number  * Street Name  \
0    1         1        No Damage           8376.0   Quail Canyon   
1    2         2  Affected (1-9%)           8402.0   Quail Canyon   
2    3         3        No Damage           8430.0  Quail Canyon    
3    4         4        No Damage           3838.0    Putah Creek   
4    5         5        No Damage           3830.0    Putah Creek   

  * Street Type (e.g. road, drive, lane, etc.)  \
0                                         Road   
1                                         Road   
2                                         Road   
3                                         Road   
4                                         Road   

  Street Suffix (e.g. apt. 23, blding C)   * City State  Zip Code  ...  \
0                                    NaN  Winters    CA       NaN  ...   
1                                    NaN  Winters    CA       NaN  ...   
2                                    NaN  Winters    CA       NaN  ...   
3                                    NaN  Winters    CA       NaN  ...   
4                                    NaN  Winters    CA       NaN  ...   

  Fire Name (Secondary) APN (parcel) Assessed Improved Value (parcel)  \
0                 Quail    101090290                         510000.0   
1                 Quail    101090270                         573052.0   
2                 Quail    101090310                         350151.0   
3                 Quail    103010240                         134880.0   
4                 Quail    103010220                         346648.0   

  Year Built (parcel)                    Site Address (parcel)  \
0              1997.0  8376 QUAIL CANYON RD VACAVILLE CA 95688   
1              1980.0  8402 QUAIL CANYON RD VACAVILLE CA 95688   
2              2004.0  8430 QUAIL CANYON RD VACAVILLE CA 95688   
3              1981.0     3838 PUTAH CREEK RD WINTERS CA 95694   
4              1980.0     3830 PUTAH CREEK RD WINTERS CA 95694   

                               GLOBALID   Latitude   Longitude            x  \
0  e1919a06-b4c6-476d-99e5-f0b45b070de8  38.474960 -122.044465 -13585927.70   
1  b090eeb6-5b18-421e-9723-af7c9144587c  38.477442 -122.043252 -13585792.71   
2  268da70b-753f-46aa-8fb1-327099337395  38.479357 -122.044585 -13585941.01   
3  64d4a278-5ee9-414a-8bf4-247c5b5c60f9  38.487313 -122.015115 -13582660.52   
4  1b44b214-01fd-4f06-b764-eb42a1ec93d7  38.485636 -122.016122 -13582772.60   

             y  
0  4646740.750  
1  4647093.599  
2  4647366.034  
3  4648497.399  
4  4648258.826  

[5 rows x 47 columns]

# Exploratory Data Analysis (EDA)

In [19]:
# Understanding the basic struction of the dataset
df.info()
df.shape

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100230 entries, 0 to 100229
Data columns (total 47 columns):
 #   Column                                                        Non-Null Count   Dtype  
---  ------                                                        --------------   -----  
 0   _id                                                           100230 non-null  int64  
 1   OBJECTID                                                      100230 non-null  int64  
 2   * Damage                                                      100230 non-null  object 
 3   * Street Number                                               95810 non-null   float64
 4   * Street Name                                                 94744 non-null   object 
 5   * Street Type (e.g. road, drive, lane, etc.)                  87033 non-null   object 
 6   Street Suffix (e.g. apt. 23, blding C)                        44148 non-null   object 
 7   * City                                                  

(100230, 47)

**Dataset Columns Describtion**

OBJECTID: A unique identifier for each record in the dataset.

DAMAGE: Indicates the level of fire damage to the structure (e.g., "No Damage", "Affected (1-9%)").

STREETNUMBER: The street number of the impacted structure.

STREETNAME: The name of the street where the impacted structure is located.

STREETTYPE: The type of street (e.g., "Road", "Lane").

STREETSUFFIX: Additional address information, such as apartment or building numbers (if applicable).

CITY: The city where the impacted structure is located.

STATE: The state abbreviation (e.g., "CA" for California).

ZIPCODE: The postal code of the impacted structure.

CALFIREUNIT: The CAL FIRE unit responsible for the area.

COUNTY: The county where the impacted structure is located.

COMMUNITY: The community or neighborhood of the structure.

INCIDENTNAME: The name of the fire incident that impacted the structure.

APN: The Assessor’s Parcel Number (APN) of the property.

ASSESSEDIMPROVEDVALUE: The assessed value of the improved property (e.g., structures, not just land).

YEARBUILT: The year the structure was built.

SITEADDRESS: The full address of the property, including city, state, and ZIP code.

GLOBALID: A globally unique identifier for each record.

Latitude: The latitude coordinate of the structure’s location.

Longitude: The longitude coordinate of the structure’s location.

UTILITYMISCSTRUCTUREDISTANCE: The distance between the main structure and any utility or miscellaneous structures (if recorded).

FIRENAME: An alternative or secondary name for the fire incident.

geometry: A geospatial representation of the location in a point format (e.g., "POINT (-13585927.697 4646740.750)").

In [21]:
# Basic statistical Measures
df.describe().transpose()

count          mean  \
_id                                       100230.0  5.011550e+04   
OBJECTID                                  100230.0  5.022778e+04   
* Street Number                            95810.0  3.886722e+04   
Zip Code                                   47429.0  4.630970e+04   
# Units in Structure (if multi unit)       31184.0  4.332991e-01   
# of Damaged Outbuildings < 120 SQFT       31085.0  8.756635e-02   
# of Non Damaged Outbuildings < 120 SQFT   31073.0  1.215203e-01   
Assessed Improved Value (parcel)           94195.0  7.337022e+05   
Year Built (parcel)                        69812.0  1.672284e+03   
Latitude                                  100230.0  3.832295e+01   
Longitude                                 100230.0 -1.211793e+02   
x                                         100230.0 -1.348962e+07   
y                                         100230.0  4.629002e+06   

                                                   std           min  \
_id                                       2.893405e+04  1.000000e+00   
OBJECTID                                  2.910768e+04  1.000000e+00   
* Street Number                           5.271695e+06  0.000000e+00   
Zip Code                                  4.746765e+04  0.000000e+00   
# Units in Structure (if multi unit)      3.460877e+01  0.000000e+00   
# of Damaged Outbuildings < 120 SQFT      4.627289e-01  0.000000e+00   
# of Non Damaged Outbuildings < 120 SQFT  5.255796e-01  0.000000e+00   
Assessed Improved Value (parcel)          8.603013e+06  0.000000e+00   
Year Built (parcel)                       7.084518e+02  0.000000e+00   
Latitude                                  2.019086e+00  3.259255e+01   
Longitude                                 1.538342e+00 -1.237746e+02   
x                                         1.712474e+05 -1.377852e+07   
y                                         2.825063e+05  3.841346e+06   

                                                   25%           50%  \
_id                                       2.505825e+04  5.011550e+04   
OBJECTID                                  2.505825e+04  5.011550e+04   
* Street Number                           7.232500e+02  4.308500e+03   
Zip Code                                  0.000000e+00  0.000000e+00   
# Units in Structure (if multi unit)      0.000000e+00  0.000000e+00   
# of Damaged Outbuildings < 120 SQFT      0.000000e+00  0.000000e+00   
# of Non Damaged Outbuildings < 120 SQFT  0.000000e+00  0.000000e+00   
Assessed Improved Value (parcel)          5.937000e+04  1.455510e+05   
Year Built (parcel)                       1.944000e+03  1.972000e+03   
Latitude                                  3.735093e+01  3.869296e+01   
Longitude                                -1.223162e+02 -1.216003e+02   
x                                        -1.361617e+07 -1.353648e+07   
y                                         4.488135e+06  4.677785e+06   

                                                   75%           max  
_id                                       7.517275e+04  1.002300e+05  
OBJECTID                                  7.517275e+04  1.012210e+05  
* Street Number                           1.000300e+04  1.410065e+09  
Zip Code                                  9.566700e+04  9.631100e+04  
# Units in Structure (if multi unit)      0.000000e+00  6.101000e+03  
# of Damaged Outbuildings < 120 SQFT      0.000000e+00  4.000000e+01  
# of Non Damaged Outbuildings < 120 SQFT  0.000000e+00  2.000000e+01  
Assessed Improved Value (parcel)          3.109355e+05  1.220403e+09  
Year Built (parcel)                       1.987000e+03  2.022000e+03  
Latitude                                  3.976387e+01  4.199119e+01  
Longitude                                -1.205093e+02 -1.164182e+02  
x                                        -1.341503e+07 -1.295961e+07  
y                                         4.831688e+06  5.159661e+06

In [22]:
# Finding unique values
df.nunique()

_id                                                             100230
OBJECTID                                                        100230
* Damage                                                             6
* Street Number                                                  18098
* Street Name                                                     9818
* Street Type (e.g. road, drive, lane, etc.)                        27
Street Suffix (e.g. apt. 23, blding C)                            2260
* City                                                             442
State                                                                1
Zip Code                                                           206
* CAL FIRE Unit                                                     27
County                                                              50
Community                                                         1001
Battalion                                                           30
* Incident Name                                                    270
Incident Number (e.g. CAAEU 123456)                                294
Incident Start Date                                                294
Hazard Type                                                          3
If Affected 1-9% - Where did fire start?                            18
If Affected 1-9% - What started fire?                                9
Structure Defense Actions Taken                                     10
* Structure Type                                                    19
Structure Category                                                   7
# Units in Structure (if multi unit)                                25
# of Damaged Outbuildings < 120 SQFT                                10
# of Non Damaged Outbuildings < 120 SQFT                            12
* Roof Construction                                                 11
* Eaves                                                              6
* Vent Screen                                                       10
* Exterior Siding                                                   11
* Window Pane                                                        7
* Deck/Porch On Grade                                                6
* Deck/Porch Elevated                                                6
* Patio Cover/Carport Attached to Structure                          5
* Fence Attached to Structure                                        4
Distance - Propane Tank to Structure                                 6
Distance - Residence to Utility/Misc Structure &gt; 120 SQFT         6
Fire Name (Secondary)                                               23
APN (parcel)                                                     55479
Assessed Improved Value (parcel)                                 38191
Year Built (parcel)                                                157
Site Address (parcel)                                            53337
GLOBALID                                                        100230
Latitude                                                         99989
Longitude                                                        99552
x                                                                99612
y                                                               100009
dtype: int64

In [30]:
# checking missing values and duplicates
df.isnull().sum()
# missing values percentage and concatenate
missing = df.isnull().sum().sort_values(ascending=False)
percentage_of_missing = df.isnull().sum()/df.isnull().count().sort_values(ascending=False)
missing_data = pd.concat([missing, percentage_of_missing], axis=1, keys=['Missing_values', 'Percentage'])
missing_data

Missing_values  Percentage
Battalion                                                    93832    0.936167
If Affected 1-9% - What started fire?                        91214    0.910047
If Affected 1-9% - Where did fire start?                     89490    0.892846
Distance - Residence to Utility/Misc Structure ...           85874    0.856769
Fire Name (Secondary)                                        79059    0.788776
Distance - Propane Tank to Structure                         77173    0.769959
Structure Defense Actions Taken                              75760    0.755862
# of Non Damaged Outbuildings < 120 SQFT                     69157    0.689983
# of Damaged Outbuildings < 120 SQFT                         69145    0.689863
# Units in Structure (if multi unit)                         69046    0.688876
Street Suffix (e.g. apt. 23, blding C)                       56082    0.559533
Zip Code                                                     52801    0.526798
Community                                                    47338    0.472294
* City                                                       31614    0.315415
Year Built (parcel)                                          30418    0.303482
* Fence Attached to Structure                                21595    0.215454
* Street Type (e.g. road, drive, lane, etc.)                 13197    0.131667
Assessed Improved Value (parcel)                              6035    0.060212
* Street Name                                                 5486    0.054734
Site Address (parcel)                                         4639    0.046284
* Street Number                                               4420    0.044099
* Vent Screen                                                 1198    0.011953
* Eaves                                                       1112    0.011094
* Window Pane                                                  976    0.009738
* Exterior Siding                                              909    0.009069
* Roof Construction                                            628    0.006266
APN (parcel)                                                   617    0.006156
County                                                          30    0.000299
State                                                            7    0.000070
GLOBALID                                                         0    0.000000
Latitude                                                         0    0.000000
x                                                                0    0.000000
Longitude                                                        0    0.000000
_id                                                              0    0.000000
* Patio Cover/Carport Attached to Structure                      0    0.000000
* Deck/Porch Elevated                                            0    0.000000
* Deck/Porch On Grade                                            0    0.000000
OBJECTID                                                         0    0.000000
Structure Category                                               0    0.000000
* Structure Type                                                 0    0.000000
Hazard Type                                                      0    0.000000
Incident Start Date                                              0    0.000000
Incident Number (e.g. CAAEU 123456)                              0    0.000000
* Incident Name                                                  0    0.000000
* CAL FIRE Unit                                                  0    0.000000
* Damage                                                         0    0.000000
y                                                                0    0.000000

In [31]:
df.duplicated().sum()

0

In [35]:
# Finding categorical columns
df.select_dtypes(include=['object']).columns
categorical_columns = df.columns[df.nunique()<10]
for col in categorical_columns:
  print(col, df[col].unique())

* Damage ['No Damage' 'Affected (1-9%)' 'Minor (10-25%)' 'Destroyed (>50%)'
 'Major (26-50%)' 'Inaccessible']
State ['CA' nan]
Hazard Type ['Fire' 'Earthquake' 'Flood']
If Affected 1-9% - What started fire? [nan 'Unknown' 'Direct flame impingement' 'Embers' 'Radiant Heat'
 'Not Applicable' 'Bushes' 'Embers or overheated electrical motor'
 'Post on structure' '0-10']
Structure Category ['Single Residence' 'Other Minor Structure' 'Multiple Residence'
 'Nonresidential Commercial' 'Mixed Commercial/Residential'
 'Infrastructure' 'Agriculture']
* Eaves ['Unenclosed' 'Enclosed' 'Unknown' 'No Eaves' ' ' 'Not Applicable' nan]
* Window Pane ['Single Pane' 'Multi Pane' 'Unknown' 'No Windows' ' ' nan 'No Deck/Porch'
 'Radiant Heat']
* Deck/Porch On Grade ['Wood' 'Masonry/Concrete' 'No Deck/Porch' 'Unknown' 'Composite' ' ']
* Deck/Porch Elevated ['Wood' 'No Deck/Porch' 'Unknown' 'Masonry/Concrete' 'Composite' ' ']
* Patio Cover/Carport Attached to Structure ['No Patio Cover/Carport' 'Combustible' 

In [37]:
# Finding Numerical Columns
df.select_dtypes(include=['int64', 'float64']).columns

Index(['_id', 'OBJECTID', '* Street Number', 'Zip Code',
       '# Units in Structure (if multi unit)',
       '# of Damaged Outbuildings < 120 SQFT',
       '# of Non Damaged Outbuildings < 120 SQFT',
       'Assessed Improved Value (parcel)', 'Year Built (parcel)', 'Latitude',
       'Longitude', 'x', 'y'],
      dtype='object')

# My Observation

Based on my Obervaiton,

the dataset doesn't have duplicate records but it has a lot of missing values in both categorical and numerical columns.

**Duplicates:** Nil

**Missing Values:** Present in both numerical and categorical column, so we need encode the values inorder to perform machine learning models.

**Target Variable:** as per the objective to predict the Damages happened from Wildfire, i am setting 'Damage' column as Target variable.
